### PR TITLE
Add contribution guidelines

### LINKED ISSUE #107: Define contribution guidelines
Guidelines should define:
- Branch naming convention
- Commit naming convention
- Contents of PR
- Contents of issue

### PR DESCRIPTION
**Pull request by akshay-ap**
Fixes #107

Changes in PR:
- Create `CONTRIBUTING.md` file
- Add convention for branch naming, commit messages
- Add guidelines on issues and PR contents

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Contributing to Safe Core Protocol
+
+Even if you don't write code, there are many ways you can help improve the project.
+
+Here are some important resources:
+
+  * [Safe docs](https://docs.safe.global) tells you about Safe
+  * [Safe{Core} Protocol Specification](https://github.com/safe-global/safe-core-protocol-specs) tells you about the protocol specifications.
+
+## Submitting issues
```
Comment (rmeissner) [inline CONTRIBUTING.md:10]: Should we have issue templates for this?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  * Explain in detail the purpose and how it should work.
+  * Examples of how it should be implemented along with alternatives are welcome.
+
+If you are submitting a bug report, please:
+  * Include the steps to reproduce the bug.
+  * Include the expected behavior.
+  * Possible solutions to fix the bug are welcome.
+
+## Submitting changes
```
Comment (rmeissner) [inline CONTRIBUTING.md:22]: Should we have a PR template for this?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+Please send a [GitHub Pull Request to safe-core-protocol repository](https://github.com/safe-global/safe-core-protocol) with a clear description of the proposed changes. Each pull request should be associated with an issue and should be made against the `main` branch.
```
Comment (rmeissner) [inline CONTRIBUTING.md:24]: Do we expect external contributors to open an issue if it does not exist so far? Or are the branching conventions primarily for the core maintainers of the repo (so the devs that have write access to this repo).

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+- For a new feature, use `feature-<issue-number>-short-description`
+- For a bug fix, use `fix-<issue-number>-short-description`
+
+Always write a clear log message for your commits. The commit message should look like this:
+
+    $ git commit -m "[#<issue-number>] A brief summary of the commit"
+
+Steps to be taken before submitting a pull request to be considered for review:
+- Documentation reflects the changes
+- Make sure test coverage is 100%
```
Comment (rmeissner) [inline CONTRIBUTING.md:37]: I would say "is close to"

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  * [Safe{Core} Protocol Specification](https://github.com/safe-global/safe-core-protocol-specs) tells you about the protocol specifications.
+
+## Submitting issues
+
+If you are submitting a feature request, please:
+
+  * Explain in detail the purpose and how it should work.
+  * Examples of how it should be implemented along with alternatives are welcome.
+
+If you are submitting a bug report, please:
```
Comment (rmeissner) [inline CONTRIBUTING.md:17]: Should we link to our bug bounty here?

**Review by rmeissner — APPROVED** [review]
There is nothing really blocking this, but we could add more info and templates to make it more complete

File: CONTRIBUTING.md (hunk 3 above)
Comment (akshay-ap) [reply CONTRIBUTING.md:25]: Yes, I think there should be issue for every PR and even external contributors should open one if does not exist.

**Pull request merged**